### PR TITLE
Breaking: drop node v10/v13/v15 (refs eslint/eslint#14023)

### PR DESCRIPTION
**Pull request by aladdin-add**

### CODE REVIEW

**Review by snitin315 — COMMENTED** [review]
It looks like we have a small lint issue.

**Review by snitin315 — APPROVED** [review]
LGTM 🚀

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node: [16.x, 15.x, 14.x, 13.x, 12.x, 10.x, "10.12.0"]
+        node: [16.x, 14.x, 12.x, "12.22.0"]
```
Comment (ljharb) [inline .github/workflows/ci.yml:25]:
```suggestion
        node: [16.x, 14.x, "14.17.0", 12.x, "12.22.0"]
```

File: package.json [hunk 2]
```diff
@@ -58,6 +58,6 @@
     "strip-json-comments": "^3.1.1"
   },
   "engines": {
-    "node": "^10.12.0 || >=12.0.0"
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
```
Comment (ljharb) [inline package.json:61]: beware the >=, that will force v17 etc whether you intend it or not

**Review by ljharb — CHANGES_REQUESTED** [review]

**Comment by nzakas** [comment]
Let’s make sure we sync these changes with the final changes in the eslint repo

**Review by nzakas — APPROVED** [review]

**Pull request merged**